Leo/planner cart/course links

leonardo-fan opened the pull request:
- added links back to courses in planner cart
- fixed bug where the highlighted header page didn't change when not changing via the header (useEffect using the window path address)

leonardo-fan committed:
added links to courses navigating to course-selector

leonardo-fan committed:
fixed bug where header highlighting of current page did not change

leonardo-fan committed:
Merge branch 'dev' into leo/planner-cart/course-links

Snuzzn commented on frontend/src/pages/CourseSelector/SearchCourse.jsx:
```diff
@@ -8,7 +8,7 @@ import { useDebounce } from "../../hooks/useDebounce";
 export default function SearchCourse() {
   const [value, setValue] = useState("");
-  const debouncedSearchTerm = useDebounce(value, 200);
+  const debouncedSearchTerm = useDebounce(value, 500);
```
I recently changed this to 200 to reduce the delay :).

leonardo-fan replied:
Ohh I see - I think when I merged dev it didn't take in the new changes for some reason, I'll go in to fix them up

Snuzzn reviewed (approved):
lgtm!

dependabot[bot] committed:
chore(deps): bump fastapi from 0.75.0 to 0.75.1 in /backend (#312)

Bumps [fastapi](https://github.com/tiangolo/fastapi) from 0.75.0 to 0.75.1.
- [Release notes](https://github.com/tiangolo/fastapi/releases)
- [Commits](https://github.com/tiangolo/fastapi/compare/0.75.0...0.75.1)

---
updated-dependencies:
- dependency-name: fastapi
  dependency-type: direct:production
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump hypothesis from 6.40.1 to 6.41.0 in /backend (#311)

Bumps [hypothesis](https://github.com/HypothesisWorks/hypothesis) from 6.40.1 to 6.41.0.
- [Release notes](https://github.com/HypothesisWorks/hypothesis/releases)
- [Commits](https://github.com/HypothesisWorks/hypothesis/compare/hypothesis-python-6.40.1...hypothesis-python-6.41.0)

---
updated-dependencies:
- dependency-name: hypothesis
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

dependabot[bot] committed:
chore(deps): bump zipp from 3.7.0 to 3.8.0 in /backend (#310)

Bumps [zipp](https://github.com/jaraco/zipp) from 3.7.0 to 3.8.0.
- [Release notes](https://github.com/jaraco/zipp/releases)
- [Changelog](https://github.com/jaraco/zipp/blob/main/CHANGES.rst)
- [Commits](https://github.com/jaraco/zipp/compare/v3.7.0...v3.8.0)

---
updated-dependencies:
- dependency-name: zipp
  dependency-type: direct:production
  update-type: version-update:semver-minor
...

Signed-off-by: dependabot[bot] <support@github.com>

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Peedee2002 committed:
made a few misc cleanup changes and added a script to re-run the processors for archived years. (#309)

made a few misc cleanup changes and added a script to re-run the processors for archived years.

leonardo-fan committed:
reverted erroneous merge from past dev b3a0898356e6687fc4a8c82b8761f2aedd4df9d2

leonardo-fan committed:
updated package-lock.json

leonardo-fan committed:
newline
